src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,405 @@
+
+        //Return string removing the letter between "z" and "p"
+        return strZP;
+      } else {
+        return strZP;
+      }
+    }
+    /**
+    *Write a method longestWord(String filenametxt) such that given the name of a file filenametxt that *contains a single word on each line, returns the longest word in the file.
+    *
+    * @param filename - the file that contains all the words
+    * @return strWord - the string that is the longest in the file
+    * @author: E. Aroozoo
+    */
+
+    public static String longestWord(String filenametxt) throws IOException {
+      BufferedReader filename = new BufferedReader(new FileReader(filenametxt));
+
+      //Variables
+      String strWord;
+      String strLongestWord;
+      int intWordLength;
+      int intLongestWordLength;
+
+      //Setting Variables (Putting the data from the file into the variable)
+      strWord = filename.readLine();
+      strLongestWord = filename.readLine();
+
+
+      //Comparing the next word to see if it is longer than the current word
+      while(strWord != null) {
+        intWordLength = strWord.length();
+        intLongestWordLength = strLongestWord.length();
+          if(intWordLength >= intLongestWordLength) {
+            strLongestWord = strWord;
+        }
+        strWord = filename.readLine();
+      }
+
+      //Closing File
+      filename.close();
+      //Returning Longest Word
+      return strLongestWord;
+    }
+    /**
+    *For each multiple of 10 in the given array, change all the values following it to be that multiple of *10, until encountering another multiple of 10. So {2, 10, 3, 4, 20, 5} yields {2, 10, 10, 10, 20, 20}.
+    *
+    * @param nums - the array containing the set of numbers
+    * @return nums - the updated array containing the same multiples of 10
+    * @author: E. Aroozoo
+    */
+    public static int[] tenRun(int[] nums) {
+
+    //Variables
+    int i;
+    int intMultiple;
+
+    //Setting Variable
+    intMultiple = 1;
+
+      for(i = 0; i < nums.length; i++) {
+
+        //We know it is a multiple of 10 if the remainder is 0 when divided by 10
+        if(nums[i] % 10 == 0) {
+          intMultiple = nums[i];
+
+        //If it is not a multiple of 10, replace it with the multiple of 10
+        } else if(nums[i] % 10 != 0 && intMultiple != 1) {
+          nums[i] = intMultiple;
+        }
+      }
+
+    //Returning array with the new values
+    return nums;
+    }
+    /**
+    *Given two arrays of ints sorted in increasing order, outer and inner, return true if all of the numbers in inner appear in outer. The best solution makes only a single "linear" pass of both arrays, taking advantage of the fact that both arrays are already in sorted order
+    *
+    * @param inner - the inner array
+    * @param outer - the outer array 
+    * @return true/false if numbers in inner array are in outer array
+    * @author: E. Aroozoo
+    */
+    public static boolean linearIn(int[] outer, int[] inner) {
```

Comment:
@EngelbertEA 
`Test: Utility.linearIn(new int[]{1}, new int[]{1})` --> `java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at gr11review.part2.Utility.linearIn(Utility.java:149)`